app/src/main/java/io/gnosis/safe/ui/safe/terms/TermsChecker.kt

```diff
@@ -0,0 +1,21 @@
+package io.gnosis.safe.ui.safe.terms
+
+import pm.gnosis.svalinn.common.PreferencesManager
+import pm.gnosis.svalinn.common.utils.edit
+import javax.inject.Inject
+
+class TermsChecker @Inject constructor(
+    private val preferencesManager: PreferencesManager
+) {
+    fun setTermsAgreed(value: Boolean) {
```

Comment:
As you can see in the `SafeRepository`, any interaction with sharedPreferences is assumed to be asynchronous. I would ask you to add `suspend` to both this and the `getTermsAgreed` method.

Reply:
For writing to SharePreferences that is definitely not necessary because `SharedPreferences.edit()` uses `Editor.apply()`. For reading it's not clear, whether it is recommended. Some say so, some say so. 

If I use suspend there, then I also have to inherit from BaseStateViewModel and use AppDispatchers, right?

Reply:
`suspend` on `setActiveSafe()` is redundant (according to Android Studio) and on `getActiveSafe()` it is only necessary because of the (network) call to `getSafeBy()`. Not because of the interaction with the shared preferences.